network_diffusion/tpn/tpnetwork.py

```diff
@@ -1,3 +1,23 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 by Yuxuan Qiu. All Rights Reserved.
+#
+# This file is part of Network Diffusion.
+#
+# Network Diffusion is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.
+#
+# Network Diffusion is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the  GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# Network Diffusion. If not, see <http://www.gnu.org/licenses/>.
+# =============================================================================
+
 """A script where a temporal network is defined."""
 
 from typing import Any, Dict, List, Optional
```

Comment:
@nurek-mateusz - I think it will be good to integrate cogsnet function with the TemporalNetwork class